include/datadog/opentracing.h

```diff
@@ -32,7 +35,24 @@ struct TracerOptions {
   std::string operation_name_override = "";
 };
 
+// TracePublisher exposes the data required to publish traces to the
+// Datadog Agent.
+class TracePublisher {
+ public:
+  TracePublisher() {}
+  virtual ~TracePublisher() {}
+
+  // Returns the Datadog Agent endpoint that traces should be published to.
+  virtual const std::string path() = 0;
+  virtual std::size_t pendingTraces() = 0;
+  virtual void clearTraces() = 0;
+  virtual const std::map<std::string, std::string> headers() = 0;
+  virtual const std::string payload() = 0;
+};
+
 std::shared_ptr<ot::Tracer> makeTracer(const TracerOptions &options);
+std::tuple<std::shared_ptr<ot::Tracer>, std::shared_ptr<TracePublisher>> makeTracerAndPublisher(
```

Comment:
Might be good to add comments so anyone looking here would know why they'd want to use one method or the other.